discovery/gh-discovery.go

```diff
@@ -0,0 +1,257 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	"gopkg.in/src-d/go-errors.v1"
+
+	"github.com/google/go-github/github"
+	"github.com/jpillora/backoff"
+)
+
+var (
+	// ErrEndpointsNotFound is the returned error when couldn't find
+	// endpoints for a certain repository.
+	ErrEndpointsNotFound = errors.NewKind("endpoinds not found for %s")
+
+	// ErrNewRepositoriesNotFound is returned when there aren't new
+	// repositories in the organization.
+	ErrNewRepositoriesNotFound = errors.NewKind(
+		"couldn't find new repositories")
+
+	// ErrRateLimitExceeded is returned when the api rate limit is reached.
+	ErrRateLimitExceeded = errors.NewKind("rate limit requests exceeded")
+
+	// ErrDiscoveryStopped is returned when a discovery has been stopped.
+	ErrDiscoveryStopped = errors.NewKind("discovery stopped")
+
+	// ErrDiscoveryStop is returned when a discovery fails on Stop.
+	ErrDiscoveryStop = errors.NewKind("discovery failed on stop")
+
+	// ErrAdvertiseTimeout is returned when an advertise functions exceeds
+	// the timeout.
+	ErrAdvertiseTimeout = errors.NewKind("advertise repositories timeout")
+)
+
+// AdvertiseGHRepositoriesFn is used by a GHDiscovery to notify that a new
+// repository has been discovered.
+type AdvertiseGHRepositoriesFn func(context.Context, []*github.Repository) error
+
+// GHDiscoveryOpts represents configuration options for a GHDiscovery.
+type GHDiscoveryOpts struct {
+	AdvertiseTimeout time.Duration
+	SkipForks        bool
+	WaitNewRepos     bool
+	WaitOnRateLimit  bool
+	StopTimeout      time.Duration
+	MaxJobBuffer     int
+	BatchSize        int
+}
+
+// GHDiscovery will retrieve the information for all the repositories for the
+// given GHRepositoriesIterator.
+type GHDiscovery struct {
```

Comment:
Why not `GitHub`? It's already in `discovery` package. Same for the file name `discovery/github.go`.